Clinical-Genomics/chanjo2
# Fixed SQL models and added a test

### This PR adds | fixes:
Fix #4: 
- Improved sql database table.
- Separated sql_models from pydantic_models (otherwise it's confusing) and put them on different files named accordingly
- Moved code that will be shared by most tests into fixtures (conftest.py file)
- Renamed SQL table `Individuals` to `Samples` 
- Renamed table `Region` to `Interval`
- Added table `Cases`
- Added new table `Tags` -> will contain data like a gene in a certain genome build, or a custom genomic interval of a certain build
- Added new table `interval_tag`, an in-between table to reflect the relationship between `Intervals` and `Tags`. (for instance one genomic interval can have many tags - APC build 38, genome build 38, etc - but also one tag can have many genomic intervals -like genome build 38- --> Many to many relationship. There must be other ways, but this one looks flexible to me..
-  Added test to check that database schema is created when app lauches

Proposed database schema below. I'm sure it can be improved, but it will help a lot starting populating a draft with demo data.

<img width="1413" alt="image" src="https://user-images.githubusercontent.com/28093618/214806449-680ad190-332a-44ea-9349-48fb5d405859.png">


**How to prepare for test**:
- Install the repo locally conda env + poetry install

### How to test:
- Launch the app using the command `uvicorn src.chanjo2.main:app`
- Run tests locally and make sure that the database file created is reflecting the schema above



### Expected outcome:
- [x] the app should work as before
- [x] Tests should pass, both locally and in GitHub action

### Review: 

This [version](https://semver.org/) is a:
